crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -351,24 +351,43 @@ where
 
     /// Create a new package
     pub fn new_package<'p>(
-        &self,
+        &mut self,
         modules: Vec<move_binary_format::CompiledModule>,
         dependencies: impl IntoIterator<Item = &'p MovePackage>,
         version: Option<SequenceNumber>,
-    ) -> Result<Object, ExecutionError> {
+    ) -> Result<ObjectID, ExecutionError> {
         // wrap the modules in an object, write it to the store
-        Object::new_package(
+        let object = Object::new_package(
             modules,
             version.unwrap_or(OBJECT_START_VERSION),
             self.tx_context.digest(),
             self.protocol_config.max_move_package_size(),
             dependencies,
-        )
+        )?;
+        let object_id = object.id();
+        self.new_packages.push(object);
+        Ok(object_id)
     }
 
-    /// Add the package to the new package published
-    pub fn add_package(&mut self, package_obj: Object) {
-        self.new_packages.push(package_obj)
+    /// Upgrades the package: checks compatibility and increments the version number.
```

Comment:
Where is compatibility checked here?

Reply:
Ah damn, this is an old comment from when upgrading a package would run the compatibility checker as well.  I'll update.